app/models/game.rb

```diff
@@ -60,6 +61,58 @@ def initialize_board
     King.create(is_black: true, x_position: 4, y_position: 7, game_id: id, user_id: user_id_black, status: 'active')
     Queen.create(is_black: true, x_position: 3, y_position: 7, game_id: id, user_id: user_id_black, status: 'active')
   end
+
   # rubocop:enable Metrics/AbcSize
   # rubocop:enable MethodLength
+
+  # returns true if game is in check
+  def black_in_check?(x, y)
+    # identify the black kings on the board
+    King.where(is_black: true, x_position: x, y_position: y)
+    # selects white pieces and looks to see if
+    # they have an open route to the king
+    pieces.where(is_black: false).each do |piece|
```

Comment:
I don't think this would check to make sure the piece is actually active correct?